spec/factories/works.rb

```diff
@@ -27,7 +28,7 @@
       keywords_count { 3 }
     end
 
-    keywords { Array.new(keywords_count) { association(:keyword) } }
+    keywords { build_list(:keyword, keywords_count) }
```

Comment:
Nice.

Reply:
The other way fails as it tries to persist the keyword before persisting the work.